apps/dashr-yield-curve/app.R

```diff
@@ -0,0 +1,428 @@
+                 how much inflation there will be, for example, and how healthy growth will
+                 be over the years ahead - all embodied in the price of money today,
+                 tomorrow and many years from now. "), style=l.(width='120%'))), 
+  
+  l.(htmlDiv(dccMarkdown(className='six columns', children= "#### Where we stand"), style=l.(width='120%')), #htmlBr(),
+     htmlDiv( dccMarkdown(className='six columns', children="On Wednesday, both short-term and long-term rates were lower than they have
+    been for most of history - a reflection of the continuing hangover
+    from the financial crisis. The yield curve is fairly flat, which is a sign that investors expect
+    mediocre growth in the years ahead."), style=l.(width='120%'))), 
+  l.(htmlDiv(dccMarkdown(className='six columns', children=' #### Deep in the valley'),style=l.(width='120%')), #htmlBr(),
+     htmlDiv(dccMarkdown(className='six columns', children='In response to the last recession, the Federal Reserve has kept short-term
+    rates very low - near zero - since 2008. (Lower interest rates stimulate
+    the economy, by making it cheaper for people to borrow money, but also
+    spark inflation.) Now, the Fed is getting ready to raise rates again, possibly as early as
+    June.'), style=l.('width'='120%'))), 
+  l.(htmlDiv(dccMarkdown(className='six columns', children=' #### Last time, a puzzle'), style=l.(width='120%')), #htmlBr(),
+     htmlDiv(dccMarkdown(className='six columns', children='The last time the Fed started raising rates was in 2004. From 2004 to 2006,
+    short-term rates rose steadily. But long-term rates did not rise very much. The Federal Reserve chairman called this phenomenon a "conundrum," and it
+    raised questions about the ability of the Fed to guide the economy.
+    Part of the reason long-term rates failed to rise was because of strong
+    foreign demand.') , style=l.('width'='120%'))), 
+  l.(htmlDiv(dccMarkdown(className='six columns', children='#### Long-term rates are low now, too'), style=l.(width='120%')), #htmlBr(),
+     htmlDiv(dccMarkdown(className='six columns', children= 'Foreign buyers have helped keep long-term rates low recently, too - as have
+    new rules encouraging banks to hold government debt and expectations that
+    economic growth could be weak for a long time. The 10-year Treasury yield was as low as it has ever been in July 2012 and
+    has risen only modestly since.Some economists refer to the economic pessimism as "the new normal."'), style=l.('width'='120%'))), 
+  l.(htmlDiv(dccMarkdown(className='six columns', children=' #### Long-term rates are low now, too'),style=l.(width='120%') ), #htmlBr(),
+     htmlDiv(dccMarkdown(className='six columns', children= 'Here is the same chart viewed from above.'), style=l.(width='120%')))
+)
+
+##INITALIZE THE GRAPH
+graph <- DM %>% .[,dim(.)[2]:1] %>% plot_ly(x=rev(labels), y=base::as.Date(dates, format='%m/%d/%Y'),z=., type='surface', 
```

Comment:
`plotly::plot_ly(x=rev(labels), y=as.Date(dates, format='%m/%d/%Y'),z=., type='surface'` To `plotly::`, either use it all the time you call `plotly` function, or never use it.